java/monkey/src/main/java/root/parser/Parser.java

```diff
@@ -94,14 +95,96 @@ private Statement parseLetStatement() throws ParserException {
         return letStatement;
     }
 
+    private ExpressionStatement parseExpressionStatement() throws ParserException {
+        ExpressionStatement statement = new ExpressionStatement(this.currentToken);
+
+        statement.setExpression(this.parseExpression(OperatorPrecedence.LOWEST));
+
+        if (this.peekTokenIs(TokenType.SEMI)) {
+            this.nextToken();
+        }
+
+        return statement;
+    }
+
+    private void noPrefixParseFnError(TokenType type) throws ParserException {
+        throw new ParserException("No prefix parse function found for " + type);
+    }
+
+    private Expression parseExpression(OperatorPrecedence precedence) throws ParserException {
+        ParserSupplier<Expression> prefixFn = prefixParseFn();
+
+        if (prefixFn == null) {
```

Comment:
Maybe swap for `Objects.requireNonNullElseThrow()`?

Reply:
It does look better, but sadly Objects.requireNonNull and it's variants can only throw NullPointerException, but Iĺl inline the exception.